src/bda/plone/orders/common.py

```diff
@@ -521,13 +532,39 @@ def salaried(self, value):
             'Abstract OrderState does not implement salaried.setter')
 
     def update_item_stock(self, booking, old_state, new_state):
+        """Change stock according to transition. See table in transitions.py
+        """
         if old_state == new_state:
             return
         # XXX: fix stock item available??
-        if new_state == ifaces.STATE_NEW:
-            self.decrease_stock(booking)
-        if new_state == ifaces.STATE_CANCELLED:
-            self.increase_stock(booking)
+        if old_state == ifaces.STATE_NEW:
+            if new_state == ifaces.STATE_CANCELLED:
+                self.increase_stock(booking)
+            else:
```

Comment:
I kept the else paths with their pass statement just to make it explicit, that in that case, nothing else happens.
In this specific case, for a booking with STATE_NEW, the stock as already decreased. Only cancel sets that booking free and the stock can be increased with that item.